### PR TITLE
Summary error message

### PR DESCRIPTION
**Pull request by frederike-ramin**
# Short Description
- For accessibility reasons we want the errors in the flash message. As this is quite difficult, we now have the number of errors in the flash message instead which is settled with Nadine. She did not have a chance to look at the message itself though. This comes from me and I am not too happy with it. :D But probably better than no message?
- also, we previously showed the error message twice: once for post, once for the following get request
- This solves two tickets in one: https://steuerlotse.atlassian.net/browse/STL-1094 and https://steuerlotse.atlassian.net/browse/STL-1002

# Changes
- Prevent flashing two times
- Made MandatoryFieldMissingValidationError display the number of errors by transforming its message into a function

# Feedback
- Do you see a way to check or at least flash only on GET *or* POST in a better way? I tried to check if the same message was already queued to flash but once you check the messages (`get_flashed_messages`) they are removed.
- Do you think it's necessary to test that the number of errors is included in the error message? If so, I don't really know how to do that. Would probably just patch the `_l()` function and check its arguments?
- Do you have a better idea for the message itself?

Sorry for the big feedback block, I'm just not very certain about these changes.

### CODE REVIEW

File: webapp/app/model/form_data.py [hunk 1]
```diff
@@ -160,12 +160,13 @@ class InputDataInvalidError(ValueError):
...
     """Raised in case of a mandatory field missing"""
-    message = _l('form.lotse.input_invalid.mandatory_field_missing')
-
     def __init__(self, missing_fields=None):
         super().__init__()
         self.missing_fields = missing_fields
 
+    def get_message(self):
+        return _l('form.lotse.input_invalid.mandatory_field_missing', num_errors=len(self.missing_fields))
+
```
Comment (JulianRoesner) [inline webapp/app/model/form_data.py:169]: You could just test this function by setting the missing fields and then checking that the message contains the number of errors. However, I do not think it is that important to do this.

File: webapp/app/forms/flows/lotse_flow.py [hunk 2]
```diff
@@ -247,7 +247,9 @@ def _handle_specifics_for_step(self, step, render_info, stored_data):
                 self._validate_mandatory_fields(stored_data)
             except MandatoryFieldMissingValidationError as e:
                 app.logger.info(f"Mandatory est fields missing: {e.missing_fields}", exc_info=True)
-                flash(e.message, 'warn')
+                # prevent flashing the same message two times
+                if request.method == 'GET':
```
Comment (JulianRoesner) [inline webapp/app/forms/flows/lotse_flow.py:251]: No, I think we'll need that conditional at some point anyway.

File: webapp/app/translations/de/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -2262,15 +2262,17 @@ msgstr ""
 "Spenden und Mitgliedsbeiträge für steuerbegünstigte Zwecke sind nur auf "
 "Anforderung des Finanzamts durch eine Bestätigung nachzuweisen. "
 
-#: app/model/form_data.py:163
+#: app/model/form_data.py:168
 msgid "form.lotse.input_invalid.mandatory_field_missing"
-msgstr "Bitte füllen Sie alle notwendigen Felder aus."
+msgstr ""
+"Bitte füllen Sie alle notwendigen Felder aus. Es fehlen noch "
+"%(num_errors)s Felder."
```
Comment (JulianRoesner) [inline webapp/app/translations/de/LC_MESSAGES/messages.po:2269]: I think I would prefer something like: `Bitte füllen Sie alle notwendigen Felder aus. Es sind noch %(num_errors)s notwendige Felder nicht ausgefüllt.`

**Review by JulianRoesner — APPROVED** [review]
LGTM!

File: webapp/app/model/form_data.py (hunk 1 above)
Comment (frederike-ramin) [reply webapp/app/model/form_data.py:169]: Won't do it then^^

**Pull request merged**